lucet-wasi/src/main.rs

```diff
@@ -158,7 +178,15 @@ fn run(config: Config) {
     lucet_wasi::hostcalls::ensure_linked();
     let exitcode = {
         // doing all of this in a block makes sure everything gets dropped before exiting
-        let module = DlModule::load(&config.lucet_module).expect("module can be loaded");
+        let pk = match (config.verify, config.pk_path) {
+            (false, _) => None,
+            (true, Some(pk_path)) => {
+                Some(PublicKey::from_file(pk_path).expect("public key can be loaded"))
+            }
+            (true, None) => panic!("signature verification requires a public key"),
+        };
+        let module =
+            DlModule::load_and_verify(&config.lucet_module, pk).expect("module can be loaded");
```

Comment:
Usage/style nit: I'm pretty uncomfortable with `load_and_verify` possibly _not_ verifying (eg when pk is `None`) - I think this function makes sense as a helper internal to `DlModule`, but for a public function I think it's worth being a stickler about making the caller very aware they are (or are not) verifying at load.

Reply:
Nod. I'm going to explicitly call `load` or `load_and_verify` instead.

Reply:
Done.